### PR TITLE
Fix issue #7438 (Crash on Mac Mavericks in traffic light code)

### PR DESCRIPTION
**Pull request by prksingh**
Fixing trafficlightbutton crash https://github.com/adobe/brackets/issues/7438 in Brackets-shell. The crash is UTR so this is not a confirmed fix.

### CODE REVIEW

**Comment by JeffryBooher** [comment]
I looked at it and the changes look OK by me but @bobeast should have a look if he has time.  Also I need to test it and @redmunds should test it on Yosemite.

**Comment by JeffryBooher** [comment]
Fixes https://github.com/adobe/brackets/issues/7438

**Comment by redmunds** [comment]
This branch seems to revert to CEF 1547. Is that right?

**Comment by redmunds** [comment]
I can build and run Brackets on Yosemite. I did Save As a couple times and did not see any crash.

File: appshell/TrafficLightButton.mm [hunk 1]
```diff
@@ -70,21 +182,21 @@ - (void)setup {
...
-        dirtyInactive = [NSImage imageNamed:[NSString stringWithFormat:@"window-%@-dirty-inactive",buttonName]];
-        dirtyHover = [NSImage imageNamed:[NSString stringWithFormat:@"window-%@-dirty-hover",buttonName]];
-        dirtyPressed = [NSImage imageNamed:[NSString stringWithFormat:@"window-%@-dirty-pressed",buttonName]];
+        self.dirtyActive = [NSImage imageNamed:[NSString stringWithFormat:@"window-%@-dirty-active",buttonName]];
+        self.dirtyInactive = [NSImage imageNamed:[NSString stringWithFormat:@"window-%@-dirty-inactive",buttonName]];
+        self.dirtyHover = [NSImage imageNamed:[NSString stringWithFormat:@"window-%@-dirty-hover",buttonName]];
+        self.dirtyPressed = [NSImage imageNamed:[NSString stringWithFormat:@"window-%@-dirty-pressed",buttonName]];
     }
-
+
```
Comment (ingorichter) [inline appshell/TrafficLightButton.mm]: nit: please remove the trailing spaces

**Comment by nethip** [comment]
@prksingh Good catch and change! But this seems to be lot of change. How about just changing to just `active = [NSImage imageNamed:[NSString stringWithFormat:@"window-%@-active",buttonName] retain];` for all the variables instead of these changes. I am afraid too many releases and retains are going to put app to an inconsistent state. And also `NSImage` returned by `[NSImage imageNeeded]` is `autorelelease`ed and so the OS is not going to release it as long as some one is holding on to it's reference, unless someone calls class `[NSAutoReleasePool release]`. Nevertheless, this is a safe change as long as the no of lines of code is put to minimum.

I think the actual culprit here is `[NSImage imageNeeded]` itself. This could be easily substituted with other ways of loading `NSImage` where we are responsible for managing memory. So that way we could just rule out any memory issues, if any.

**Comment by JeffryBooher** [comment]
@prksingh also @redmunds ran into an issue where I think you just need to merge master into this branch so it can be tested.  It may have been based on an older branch because I don't see the change  to revert CEF expressly made.

**Comment by prksingh** [comment]
@nethip I can remove the setters and move the retain to setup(). My only worry was in case this method gets executed multiple times we may be retaining multiple copies that could cause a leak. It's great if the `autorelease` refcounting is able to handle this. Thanks for the quick review

**Comment by prksingh** [comment]
@JeffryBooher @redmunds @ingorichter Thanks for looking at this. I have updated the branch from master and removed the trailing spaces. It have also updated the changes as requested by @nethip.

Please let me know if i missed anything else. Thanks!

**Comment by redmunds** [comment]
Looks good on Yosemite.

**Comment by nethip** [comment]
@redmunds thanks for testing it out. Is it possible for you to give it a spin on Mavericks, if you have that setup already. If not, could anyone else give it a spin on Mavericks.

**Comment by peterflynn** [comment]
Awesome to see a fix for this!  I've tweaked the title a little to be more easy to understand when looking at bug/PR lists.

**Pull request merged**